common/utils/data.utils.ts

```diff
@@ -0,0 +1,178 @@
+import * as xlsx from "xlsx";
+import { InputWord, Word, Topic, Language, LanguageCode } from "../types/types";
+
+const NON_LANGUAGE_FIELDS = [
+  "Bane",
+  "Bilde_a",
+  "Bilde_b",
+  "Bilde_c",
+  "Elementtype",
+  "Tema1",
+  "Title",
+  "Undertema1",
+  "Bokmål_nb_duplisert",
+];
+
+const databaseURL =
+  "https://devaudiobildetema.blob.core.windows.net/database/database.xlsx";
+
+const languages: Language[] = [];
+const topics: Topic[] = [];
+
+export const getLanguages = async (): Promise<Language[]> => {
+  return languages;
+};
+
+export const getLanguage = async (
+  languageCode: string
+): Promise<Language | undefined> => {
+  return languages.find((item) => item.code === languageCode);
+};
+
+export const getTopics = async (): Promise<Topic[]> => {
+  return topics;
+};
+const setTopic = (topic: Topic, map: Map<LanguageCode, Topic>) => {
+  map.set(topic.label, topic);
+  languages.forEach((language) => {
+    topic.words.set(language.code, []);
+  });
+};
+
+const findSubTopics = (json: InputWord[], topicMap: Map<string, Topic>): void => {
+  json.forEach((element: InputWord) => {
+    const isSubTopic =
+      element.Title.includes("T") &&
+      element.Bokmål_nb.toLocaleLowerCase() !==
+        element.Tema1.toLocaleLowerCase();
+    if (!isSubTopic) return;
+    const topic: Topic = {
+      id: element.Title,
+      label: element.Bokmål_nb,
+      subTopics: new Map(),
+      words: new Map(),
+    };
+    setTopic(topic, topicMap.get(element.Tema1)!.subTopics);
+  });
+}
+
+const findMainTopics = (json: InputWord[], topicMap: Map<string, Topic>): void => {
+  json.forEach((element: InputWord) => {
+    const isMainTopic =
+      element.Title.includes("T") &&
+      element.Bokmål_nb.toLocaleLowerCase() ===
+        element.Tema1.toLocaleLowerCase();
+    if (!isMainTopic) return;
+
+    if (!topicMap.has(element.Tema1)) {
+      const topic: Topic = {
+        id: element.Title,
+        label: element.Tema1,
+        subTopics: new Map(),
+        words: new Map(),
+      };
+      setTopic(topic, topicMap);
+    }
+  });
+}
+
+const findTopics = (json: InputWord[], topicMap: Map<string, Topic>) => {
```

Comment:
Perhaps `words` and `topics` are more describing names for these parameters?

```suggestion
const findTopics = (words: InputWord[], topics: Map<string, Topic>) => {
```

Reply:
topics is a global variable of type Topic[]
renamed json to inputWords.